src/convertToObject.js

```diff
@@ -3,16 +3,54 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties (see an
+ * exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitStrings = sourceString.split(';');
+
+  const splitLongStrings = splitStrings.filter(x => x.length > 4);
+
+  const stringsSplitedColon = [];
+
+  splitLongStrings.forEach(x => {
+    const arrayCopy = x.split(':');
+
+    stringsSplitedColon.push(arrayCopy[0]);
+    stringsSplitedColon.push(arrayCopy[1]);
+  });
+
+  const stringsWithoutSpaces = stringsSplitedColon.map(x => {
+    let i = 0;
+
+    while (x.charAt(i) === ' ' || x.charAt(i) === '\n') {
+      i++;
+    }
+
+    const y = x.slice(i);
+    let j = y.length;
+
+    while (y.charAt(j - 1) === ' ') {
+      j--;
+    }
+
+    return y.slice(0, j);
+  });
+
+  const result = {};
+
+  for (let i = 0; i < stringsWithoutSpaces.length - 1; i += 2) {
```

Comment:
3. Now let's simplify this function some more going back to the step 1, we could use a regular map
```js
const propertiesAndKeysArr = trimmedPropertiesAndKeysArr.map(x => x.split(':'));
```
now we get this array of two arrays with two elements`[key, value]`
![image](https://github.com/mate-academy/js_style-to-object/assets/87986955/ddc16dc5-da41-4e76-9f4d-2bb02e188258)

Reply:
We run on this the map with trim, but we have to remember that there are two elements
```js
const trimmedPropertiesAndValuesArr = propertiesAndKeysArr.map(x => [x[0].trim(), x[1].trim()]);
```
now we get something like this:
![image](https://github.com/mate-academy/js_style-to-object/assets/87986955/eba01cd1-6559-405d-b329-907fff3653a5)
now we just need to map this to an object

Reply:
However the syntax may be simplified a bit by destructuring each array in functions parameter:
```js
  const trimmedPropertiesAndValuesArr = propertiesAndKeysArr.map(([key,value]) => [key.trim(), value.trim()]);
```

Reply:
And now instead of using the for loop, we could use [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) to reduce this array into one object and immediately return it:
```js
return trimmedPropertiesAndValuesArr.reduce(
  (accumulatedObject, [key, value]) => ({ ...accumulatedObject, [key]: value }),
  {}
);
```

Reply:
Overall function now will look like this:
![image](https://github.com/mate-academy/js_style-to-object/assets/87986955/283a9a94-8282-46f3-b1e0-ff4252d5cb0f)

Reply:
This is somewhat optimal, however still not enough, we could actually take care of this in single reduce:
![image](https://github.com/mate-academy/js_style-to-object/assets/87986955/54e5284c-c1c4-4ddd-8fcb-1560feec4610)

Reply:
And go even further, use the arrow function's implicit return and do this without functions body or declaring any additional variables in it:
![image](https://github.com/mate-academy/js_style-to-object/assets/87986955/8d1dfc18-02e3-43df-9206-d1040fe44db5)